boards/opentitan/earlgrey-cw310/Makefile

```diff
@@ -20,10 +19,12 @@ include ../../Makefile.common
 # Cargo `--features`. Please see `Cargo.toml` for available options.
 ifneq ($(BOARD_CONFIGURATION),)
 	CARGO_FLAGS += --features=$(BOARD_CONFIGURATION)
-else
-	CARGO_FLAGS += --features=$(DEFAULT_BOARD_CONFIGURATION)
 endif
 
+# Build test crates with the `panic = "abort"` strategy. Tests use `unwind` by
+# default which causes incompatability with `core` built via `-Zbuild-std`.
+CARGO_FLAGS += -Zpanic-abort-tests
```

Comment:
An alternative to this would be for tests to be built using `-Z build-std=core,std,alloc,test`, which would enable virtual function elimination without breaking unwinding. We would probably have to modify the main Makefile a bit to make it ergonomic to only use this configuration for tests. I don't think any of the tests in the kernel repo currently rely on unwinding, but I know we do rely on unwinding for libtock-rs unit tests already (e.g. https://github.com/tock/libtock-rs/blob/d07eaa2df10b07668b2ed51cf993b9fd50951545/unittest/src/fake/syscalls/command_impl_tests.rs#L66)

Reply:
I looked into this briefly but had some trouble disabling atomics support while building `alloc` for `riscv32imc`. Looks like [Rust has a `no_sync`](https://github.com/rust-lang/rust/blob/3f2b2eee8f46f2252d2919d7a57bf3068d7df285/library/alloc/src/lib.rs#L241) config but I wasn't sure how to set this through `build-std-[features]`.

I'm not sure that the problem is missing `libtest` since we don't seem to link to it anyway due to using a custom `#![test_runner]` in the binary crate. Is the idea that we get Tock itself to support unwinding when building with tests, and that requires `std`?

Reply:
Hmm, I think if this looks challenging the current approach is fine. I hadn't really thought all the way through the idea of this -- my thought process was basically "hmm, don't we have tests that rely on unwinding -- wait no, that was in libtock-rs -- still, that suggests unwinding could be useful for tests -- can't we just build the whole std library with build-std?". 

Thinking a bit more about it now, libtock-rs actually only uses unwinding for on-host tests, and it seems unlikely we would have any on-target tests that would use unwinding, especially in the kernel. So I will go ahead and mark this resolved

Reply:
> I looked into this briefly but had some trouble disabling atomics support while building `alloc` for `riscv32imc`. Looks like [Rust has a `no_sync`](https://github.com/rust-lang/rust/blob/3f2b2eee8f46f2252d2919d7a57bf3068d7df285/library/alloc/src/lib.rs#L241) config but I wasn't sure how to set this through `build-std-[features]`.

I don't expect any embedded code in the `tock/tock` repository to use `alloc`, so we should be able to dodge that problem here. That is a problem for `libtock-rs`, as it should eventually have an optional dependency on `alloc`, but we can cross that bridge when we get to it.

> Thinking a bit more about it now, libtock-rs actually only uses unwinding for on-host tests, and it seems unlikely we would have any on-target tests that would use unwinding

Correct, I don't depend on unwinding in embedded code. I'm not sure we want unwinding support -- I think it has an adverse impact on code size.